_posts/2017-07-06-June.md

```diff
@@ -0,0 +1,38 @@
+---
+layout: post
+title: "Dev Diary (June 2017)"
+author: "Corey Montella"
+tags: []
+---
+
+## Preview 5 of v0.3
+
+We released a new preview of v0.3, that includes several bug fixes and some new features. For a full changelog, see the [release notes](http://incidentalcomplexity.com/2017/06/28/release-0.3p5/). Two of the new features, the `console` and `file` libraries, open Eve up to using Eve in new ways. 
+
+We've supported running Eve in headless mode (without hooks to the browser) for a while now, but it wasn't useful since most of our UI, debugging, and IO at the moment are done through the browser. The `console` library allows you to write to `sdtout` and `stderr`, so you can now pipe values generated by Eve to other applications (we still need to add the ability to pipe values _into_ Eve, but that's on our list). 
+
+Speaking of getting data into Eve, the `file` library allows you write and read files directly from the filesystem, opening up the ability to read larger datasets into Eve. We'll be shortly adding facilities in Eve that parse common data formats into records, like JSON and CSS for starters.
+
+## Progress with v0.4
+
+Many of you have asked about the status of v0.4, and we're pleased to say it's coming along nicely. For those who don't know, the v0.4 runtime is implemented in Rust, mainly for increased performance. It's similar conceptually to the v0.3 runtime (based on the research of Frank McSherry into differential dataflow), but is designed with the lessons learned from v0.3. We'll share more about performance when the runtime is feature-complete and ready for wider distribution.
+
+## Community
+
+### Community Slack
+
+We opened a [community slack channel](https://slack-signup.witheve.com/), so you can chat with and get help from us and fellow Eve developers. There's usually at least one dev hanging around the channel during working hours (Pacific time), but in the future we're planning explicit office hours where our sole purpose is to help you solve problems with your code. For now, we're keeping it casual so drop by and we can talk about Eve.
+
+### Tweve
+
+Brian Theado has been working on [Tweve](https://btheado.github.io/tweve/), which marries Eve and Tiddlywiki. Tiddlywiki is a notebook-like environment that allows you to embed rich-content in your notes. Tweve is a plugin for Tiddlywiki that allows you to embed Eve code in your notebook. One of the more interesting parts of this project is that it allows you to embed multiple Eve programs in a single document, a feature that we demoed internally but never released. We're very interested to see how far Brian takes his project!
```

Comment:
The last sentence feels off-putting. Maybe instead encourage users interested in tiddlywiki to give it a shot or contribute to it's development?